components/bt/host/bluedroid/api/include/api/esp_gap_ble_api.h

```diff
@@ -17,2687 +17,3916 @@
-    ESP_BLE_SM_IOCAP_MODE,
-    /*!< Initiator Key Distribution/Generation */
-    ESP_BLE_SM_SET_INIT_KEY,
-    /*!< Responder Key Distribution/Generation */
-    ESP_BLE_SM_SET_RSP_KEY,
-    /*!< Maximum Encryption key size to support */
-    ESP_BLE_SM_MAX_KEY_SIZE,
-    /*!< Minimum Encryption key size requirement from Peer */
-    ESP_BLE_SM_MIN_KEY_SIZE,
-    /*!< Set static Passkey */
-    ESP_BLE_SM_SET_STATIC_PASSKEY,
-    /*!< Reset static Passkey */
-    ESP_BLE_SM_CLEAR_STATIC_PASSKEY,
-    /*!< Accept only specified SMP Authentication requirement */
-    ESP_BLE_SM_ONLY_ACCEPT_SPECIFIED_SEC_AUTH,
-    /*!< Enable/Disable OOB support */
-    ESP_BLE_SM_OOB_SUPPORT,
-    /*!< Appl encryption key size */
-    ESP_BLE_APP_ENC_KEY_SIZE,
-    /*!< authentication max param */
-    ESP_BLE_SM_MAX_PARAM,
+    ESP_BLE_SM_PASSKEY = 0,             /*!< Authentication method using passkey input on the local device size 6 byte (000000-999999).
+                                             - Scenario: User inputs a numeric passkey, and the remote device verifies it to ensure a secure pairing process.
+                                             - E.g. Devices that require user input for PIN codes (6 bytes), such as smart locks or car systems. */
+    ESP_BLE_SM_AUTHEN_REQ_MODE,         /*!< Authentication mode requirements of the local device, size 1 byte (unit 8).
+                                             - Scenario: Defines the local device's need for pairing, encryption, and whether to use MITM (Man-In-The-Middle) protection or secure connections.
+                                             - See `esp_ble_auth_req_t` for details. 
+                                             - Group define for `ESP_LE_AUTH_xxx`. 
+                                             - E.g. Devices with high security needs, like financial payment devices. */
+    ESP_BLE_SM_IOCAP_MODE,              /*!< Input/output capabilities of the local device in pairing, size 1 byte (unit 8).
+                                             - Scenario: Defines whether the device supports keyboard input, display output, or has no I/O capabilities, which affects the pairing method.
+                                             - See `esp_bt_io_cap_t` for details. 
+                                             - Group define for `ESP_BT_IO_CAP_xxx`. */
+    ESP_BLE_SM_SET_INIT_KEY,            /*!< Initiator key distribution during pairing, size 1 byte (unit 8).
+                                             - Scenario: Defines the types of keys distributed by the local device during pairing. */
+    ESP_BLE_SM_SET_RSP_KEY,             /*!< Responder key distribution during pairing, size 1 byte (unit 8).
+                                             - Scenario: Defines the types of keys distributed by the local device during pairing. */
+    ESP_BLE_SM_MAX_KEY_SIZE,            /*!< Maximum encryption key size supported by the device, size 1 byte (7-16).
```

Comment:
The length of the value is 1 byte (uint8_t). The maximum key size shall be in the range 7 to 16 octets.